homeassistant/components/osoenergy/__init__.py

```diff
@@ -0,0 +1,140 @@
+            )
+        },
+    ),
+    extra=vol.ALLOW_EXTRA,
+)
+
+
+async def async_setup(hass: HomeAssistant, config: ConfigType) -> bool:
+    """OSO Energy configuration setup."""
+    hass.data[DOMAIN] = {}
+
+    if DOMAIN not in config:
+        return True
+
+    conf = config[DOMAIN]
+
+    if not hass.config_entries.async_entries(DOMAIN):
+        hass.async_create_task(
+            hass.config_entries.flow.async_init(
+                DOMAIN,
+                context={"source": config_entries.SOURCE_IMPORT},
+                data={
+                    CONF_API_KEY: conf[CONF_API_KEY],
+                },
+            )
+        )
+    return True
+
+
+async def async_setup_entry(hass: HomeAssistant, entry: ConfigEntry) -> bool:
+    """Set up OSO Energy from a config entry."""
+    if hass.data.get(DOMAIN) is None:
+        hass.data.setdefault(DOMAIN, {})
+
+    subscription_key = entry.data.get(CONF_API_KEY)
+    websession = aiohttp_client.async_get_clientsession(hass)
+    osoenergy = OSOEnergy(subscription_key, websession)
+    osoenergy_config = dict(entry.data)
+
+    osoenergy_config["options"] = {}
+    osoenergy_config["options"].update(
+        {CONF_SCAN_INTERVAL: dict(entry.options).get(CONF_SCAN_INTERVAL, 120)}
+    )
+    hass.data[DOMAIN][entry.entry_id] = osoenergy
+
+    try:
+        devices = await osoenergy.session.start_session(osoenergy_config)
+    except HTTPException as error:
+        _LOGGER.error("Could not connect to the internet: %s", error)
+        raise ConfigEntryNotReady() from error
+    except OSOEnergyReauthRequired as err:
+        raise ConfigEntryAuthFailed from err
+
+    for ha_type, oso_type in PLATFORM_LOOKUP.items():
+        device_list = devices.get(oso_type, [])
+        if device_list:
+            hass.async_create_task(
+                hass.config_entries.async_forward_entry_setup(entry, ha_type)
+            )
+
+    return True
+
+
+async def async_unload_entry(hass: HomeAssistant, entry: ConfigEntry) -> bool:
+    """Unload a config entry."""
+    unload_ok = await hass.config_entries.async_unload_platforms(entry, PLATFORMS)
+    if unload_ok:
+        hass.data[DOMAIN].pop(entry.entry_id)
+
+    return unload_ok
+
+
+def refresh_system(func):
+    """Force update all entities after state change."""
+
+    @wraps(func)
+    async def wrapper(self, *args, **kwargs):
+        await func(self, *args, **kwargs)
+        async_dispatcher_send(self.hass, DOMAIN)
+
+    return wrapper
+
+
+class OSOEnergyEntity(Entity):
+    """Initiate OSO Energy Base Class."""
+
+    def __init__(self, osoenergy, osoenergy_device):
+        """Initialize the instance."""
+        self.osoenergy = osoenergy
+        self.device = osoenergy_device
+        self.attributes = {}
+        self._unique_id = self.device["device_id"]
+
+    @property
+    def extra_state_attributes(self):
+        """Return the state attributes."""
+        return {
+            "integration": DOMAIN,
+        }
```

Comment:
This should not be in the entity attributes

Reply:
Removed the attributes